packages/account/src/Sections/Security/TwoFactorAuthentication/__tests__/digit-form.spec.tsx

```diff
@@ -0,0 +1,150 @@
+        expect(disableButton).toBeInTheDocument();
+    });
+
+    it('should display error if submits empty form', async () => {
+        render(<DigitForm {...mock_props} />);
+
+        const nameInput = screen.getByRole('textbox');
+        await userEvent.click(nameInput);
+
+        await waitFor(() => {
+            const error = screen.getByText(/Digit code is required./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should display error if user types alphanumeric characters', async () => {
+        render(<DigitForm {...mock_props} />);
+        const digitInput = screen.getByRole('textbox');
+
+        userEvent.type(digitInput, '669yi9');
+
+        await waitFor(() => {
+            const error = screen.getByText(/Digit code must only contain numbers./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should display error if user types less than 6 digits', async () => {
+        render(<DigitForm {...mock_props} />);
+        const digitInput = screen.getByRole('textbox');
+
+        userEvent.type(digitInput, '6699');
+
+        await waitFor(() => {
+            const error = screen.getByText(/Length of digit code must be 6 characters./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should display error if user types invalid OTP', async () => {
+        render(<DigitForm {...mock_props} />);
+        const digitInput = screen.getByRole('textbox');
+        const invalidOTP = '786789';
+        userEvent.type(digitInput, invalidOTP);
+
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        userEvent.click(submitButton);
+
+        await waitFor(() => {
+            const error = screen.getByText(/That's not the right code. Please try again./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should disable button when form is empty or validation fails', async () => {
+        render(<DigitForm {...mock_props} />);
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        expect(submitButton).toBeDisabled();
+
+        const digitInput = screen.getByRole('textbox');
+        userEvent.type(digitInput, '669yi9');
+        await waitFor(() => {
+            expect(submitButton).toBeDisabled();
+        });
+    });
+
+    it('should display error if response error object does not contain error code', async () => {
+        WS.authorized.accountSecurity.mockResolvedValue({
+            error: { message: '', code: '' },
+        });
+
+        render(<DigitForm {...mock_props} />);
+
+        const digitInput = screen.getByRole('textbox');
+        const invalidOTP = '786789';
+
+        userEvent.type(digitInput, invalidOTP);
+
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        userEvent.click(submitButton);
+
+        await waitFor(async () => {
```

Comment:
why do we need async here?

Reply:
queryByText is asynchronous in nature, usually we write await before screen.queryByText. However, we already mention await before waitFor(), so I used async before arrow function. Is it redundant?

Reply:
The async is added in the top level. we dont need async inside await. right?